datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -3443,6 +3525,385 @@ mod tests {
         );
     }
 
+    /// Creates a setup for chunk pruning, modeling a utf8 column "s1"
+    /// with 5 different containers (e.g. RowGroups). They have [min,
+    /// max]:
+    /// s1 ["A", "Z"]
+    /// s1 ["A", "L"]
+    /// s1 ["N", "Z"]
+    /// s1 [NULL, NULL]
+    /// s1 ["A", NULL]
+    /// s1 ["", "A"]
+    /// s1 ["", ""]
+    /// s1 ["AB", "A\u{10ffff}"]
+    /// s1 ["A\u{10ffff}\u{10ffff}\u{10ffff}", "A\u{10ffff}\u{10ffff}"]
+    fn utf8_setup() -> (SchemaRef, TestStatistics) {
+        let schema = Arc::new(Schema::new(vec![Field::new("s1", DataType::Utf8, true)]));
+
+        let statistics = TestStatistics::new().with(
+            "s1",
+            ContainerStats::new_utf8(
+                vec![
+                    Some("A"),
+                    Some("A"),
+                    Some("N"),
+                    Some("M"),
+                    None,
+                    Some("A"),
+                    Some(""),
+                    Some(""),
+                    Some("AB"),
+                    Some("A\u{10ffff}"),
```

Comment:
```suggestion
                    Some("A\u{10ffff}\u{10ffff}"),
```

Reply:
This will expose the problem we're talking about here https://github.com/apache/datafusion/pull/12978#discussion_r1817937062

Reply:
Thank you, this is great! I believe I fixed it in https://github.com/apache/datafusion/pull/12978/commits/cc0b70b8af42502a4feea0d7b544c7b1f81fafda 😄